v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandler.java

```diff
@@ -0,0 +1,940 @@
+   * the {@code valuesJson} object, parses the stringified JSON array, and returns it as a {@link
+   * Set<Object>}.
+   *
+   * @param colName - The column name used to fetch the key from {@code valuesJson}.
+   * @param valuesJson - The {@link JSONObject} containing all the key-value pairs for the current
+   *     incoming stream.
+   * @return A {@link Set<Object>} representing the parsed JSON array from the stringified JSON.
+   * @throws IllegalArgumentException If the value is not a valid stringified JSON array or cannot
+   *     be parsed.
+   */
+  public static Set<Object> handleStringifiedJsonToSet(String colName, JSONObject valuesJson) {
+    return new HashSet<>(handleStringifiedJsonToList(colName, valuesJson));
+  }
+
+  /**
+   * Converts an {@link Integer} to a {@code short} (SmallInt).
+   *
+   * <p>This method checks if the {@code integerValue} is within the valid range for a {@code
+   * smallint} (i.e., between {@link Short#MIN_VALUE} and {@link Short#MAX_VALUE}). If the value is
+   * out of range, it throws an {@link IllegalArgumentException}.
+   *
+   * @param integerValue The integer value to be converted.
+   * @return The converted {@code short} value.
+   * @throws IllegalArgumentException If the {@code integerValue} is out of range for a {@code
+   *     smallint}.
+   */
+  private static short convertToSmallInt(Integer integerValue) {
+    if (integerValue < Short.MIN_VALUE || integerValue > Short.MAX_VALUE) {
+      throw new IllegalArgumentException("Value is out of range for smallint.");
+    }
+    return integerValue.shortValue();
+  }
+
+  /**
+   * Converts an {@link Integer} to a {@code byte} (TinyInt).
+   *
+   * <p>This method checks if the {@code integerValue} is within the valid range for a {@code
+   * tinyint} (i.e., between {@link Byte#MIN_VALUE} and {@link Byte#MAX_VALUE}). If the value is out
+   * of range, it throws an {@link IllegalArgumentException}.
+   *
+   * @param integerValue The integer value to be converted.
+   * @return The converted {@code byte} value.
+   * @throws IllegalArgumentException If the {@code integerValue} is out of range for a {@code
+   *     tinyint}.
+   */
+  private static byte convertToTinyInt(Integer integerValue) {
+    if (integerValue < Byte.MIN_VALUE || integerValue > Byte.MAX_VALUE) {
+      throw new IllegalArgumentException("Value is out of range for tinyint.");
+    }
+    return integerValue.byteValue();
+  }
+
+  /**
+   * Escapes single quotes in a Cassandra string by replacing them with double single quotes.
+   *
+   * <p>This method is commonly used to sanitize strings before inserting them into Cassandra
+   * queries, where single quotes need to be escaped by doubling them (i.e., `'` becomes `''`).
+   *
+   * @param value The string to be escaped.
+   * @return The escaped string where single quotes are replaced with double single quotes.
+   */
+  private static String escapeCassandraString(String value) {
```

Comment:
This is a private method where is it being used?

Reply:
this is being used in mapping part